src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+    }
+
+    private async formatWithProfileSettings(webviewPanel: vscode.WebviewPanel) {
+        const content = await vscode.commands.executeCommand<string>("java.execute.workspaceCommand", "java.edit.stringFormatting", Example.getExample(this.exampleKind), JSON.stringify([...this.profileSettings]), this.settingsVersion);
+        webviewPanel.webview.postMessage({
+            command: "formattedContent",
+            content: content,
+        });
+    }
+
+    private async changeProfileSetting(id: string, value: string, document: vscode.TextDocument): Promise<void> {
+        const profileElement = this.profileElements.get(id);
+        if (!profileElement) {
+            // add a new setting not exist in the profile
+            if (!this.profileSettingsArray.length) {
+                return;
+            }
+            const cloneElement = this.profileSettingsArray[this.profileSettingsArray.length - 1].cloneNode();
+            const originString: string = new XMLSerializer().serializeToString(cloneElement);
+            cloneElement.setAttribute("id", id);
+            cloneElement.setAttribute("value", value);
+            const newString: string = new XMLSerializer().serializeToString(cloneElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.insert(document.uri, new vscode.Position(cloneElement.lineNumber - 1, cloneElement.columnNumber - 1 + originString.length), EOL + " ".repeat(cloneElement.columnNumber - 1) + newString);
+            vscode.workspace.applyEdit(edit);
+            cloneElement.lineNumber++;
+            this.profileElements.set(id, cloneElement);
+            this.profileSettingsArray.push(cloneElement);
+        } else {
+            // edit a current setting in the profile
+            const originSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const line: number = profileElement.lineNumber;
+            const column: number = profileElement.columnNumber;
+            await profileElement.setAttribute("value", value);
+            const newSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.replace(document.uri, new vscode.Range(new vscode.Position(line - 1, column - 1), new vscode.Position(line - 1, column - 1 + originSetting.length)), newSetting);
+            vscode.workspace.applyEdit(edit);
+        }
+    }
+
+    public async showFormatterSettingsEditor() {
+        if (!this.settingsUrl) {
+            vscode.window.showInformationMessage("No available Java Formatter Profile found, do you want to create a default profile?",
+                "Yes", "No").then((messageResult) => {
+                    if (messageResult === "Yes") {
+                        this.addFormatter();
+                    }
+                });
+        } else {
+            if (this.isRemote(this.settingsUrl)) {
```

Comment:
If thi's a remote URL, it's better to render it with read-only mode

Reply:
I'm not sure if we should
1. Create a temporary xml file for the remote profile to open the custom editor
2. Keep it in read-only mode and remain access to the preview
Instead, if we want to support the remote URL, I suggest offering an option to guide (or help) the user to download the profile and change the setting. Does this make sense?

Reply:
If the setting is set to a remote URL by user on purpose, that's indicating that the user doesn't want to change the content.

So, read-only mode is required. Meanwhile, an option to let user be able to download the file and customize it would be a plus!